ui/src/app/settings/components/project-details/project-details.tsx

```diff
@@ -473,225 +435,560 @@ export class ProjectDetails extends React.Component<RouteComponentProps<{name: s
+                                                })
+                                            )
+                                        }>
+                                        ADD DESTINATION
+                                    </button>
+                                </React.Fragment>
+                            )}
+                        </DataLoader>
+                    )}
+                    items={[]}
+                />
 
-                <h4>Blacklisted namespaced resources {helpTip('Namespace-scoped K8s API Groups and Kinds which are prohibited from being deployed')}</h4>
-                {((proj.spec.namespaceResourceBlacklist || []).length > 0 && (
-                    <div className='argo-table-list'>
-                        <div className='argo-table-list__head'>
-                            <div className='row'>
-                                <div className='columns small-3'>GROUP</div>
-                                <div className='columns small-6'>KIND</div>
-                            </div>
-                        </div>
-                        {(proj.spec.namespaceResourceBlacklist || []).map(res => (
-                            <div className='argo-table-list__row' key={`${res.group}/${res.kind}`}>
-                                <div className='row'>
-                                    <div className='columns small-3'>{res.group}</div>
-                                    <div className='columns small-6'>{res.kind}</div>
-                                </div>
-                            </div>
-                        ))}
-                    </div>
-                )) || (
-                    <div className='white-box'>
-                        <p>All namespaced-scoped resources are permitted to deploy</p>
-                    </div>
-                )}
+                <EditablePanel
+                    save={item => this.saveProject(item)}
+                    values={proj}
+                    title={<React.Fragment>CLUSTER RESOURCE ALLOW LIST {helpTip('Cluster-scoped K8s API Groups and Kinds which are permitted to be deployed')}</React.Fragment>}
+                    view={
+                        <React.Fragment>
+                            {proj.spec.clusterResourceWhitelist
+                                ? proj.spec.clusterResourceWhitelist.map((resource, i) => (
+                                      <div className='row white-box__details-row' key={i}>
+                                          <div className='columns small-4'>{resource.kind}</div>
+                                          <div className='columns small-8'>{resource.group}</div>
+                                      </div>
+                                  ))
+                                : emptyMessage('cluster resource allow list')}
+                        </React.Fragment>
+                    }
+                    edit={formApi => (
+                        <DataLoader load={() => services.clusters.list()}>
```

Comment:
Looks like `clusters` is not used here. Please remove `<DataLoader load={() => services.clusters.list()}>`